### PR TITLE
Add french and better check to see if a title is correct

### PR DESCRIPTION
**Pull request by JorisMonnet**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/blindwar/domain/game/VoiceRecognizer.kt [hunk 1]
```diff
@@ -66,7 +65,7 @@ class VoiceRecognizer : RecognitionListener {
      * @param context
      * @param editText
      */
-    fun init(context: Context, editText: EditText) {
+    fun init(context: Context, editText: EditText, language: String) {
```
Comment (marluxiaboss) [inline app/src/main/java/ch/epfl/sdp/blindwar/domain/game/VoiceRecognizer.kt:68]: Maybe the language could be an enum (small detail)

File: app/src/main/java/ch/epfl/sdp/blindwar/domain/game/VoiceRecognizer.kt [hunk 2]
```diff
@@ -76,8 +75,15 @@ class VoiceRecognizer : RecognitionListener {
                 RecognizerIntent.EXTRA_LANGUAGE_MODEL,
                 RecognizerIntent.LANGUAGE_MODEL_FREE_FORM
             )
+            speechRecognizerIntent!!.putExtra(
+                RecognizerIntent.EXTRA_LANGUAGE,
+                language
+            )
+            speechRecognizerIntent!!.putExtra(
```
Comment (marluxiaboss) [inline app/src/main/java/ch/epfl/sdp/blindwar/domain/game/VoiceRecognizer.kt:82]: Could there be an insult on the title of the song ?

File: app/src/main/java/ch/epfl/sdp/blindwar/domain/game/GameHelper.kt [hunk 3]
```diff
@@ -19,17 +15,45 @@ object GameHelper {
      * @return Boolean
      */
     fun isTheCorrectTitle(answerString: String, title: String, isVocal: Boolean): Boolean {
```
Comment (CHOCOLVT) [inline app/src/main/java/ch/epfl/sdp/blindwar/domain/game/GameHelper.kt:17]: great String error measuring function

**Review by marluxiaboss — APPROVED** [review]
Small comments on some parts of the code, otherwise I appreciate the changes you made on the code I've written.

**Review by CHOCOLVT — APPROVED** [review]
+1 for french vocal recognition
+1 for having format code accordingly to requirement

File: app/src/main/java/ch/epfl/sdp/blindwar/domain/game/GameHelper.kt [hunk 4]
```diff
@@ -19,17 +15,45 @@ object GameHelper {
...
+
+
+    /**
+     * Levenstein function to calculate distance between 2 strings
+     *
+     * @param x
+     * @param y
+     * @return the cost to go from x to y
+     */
+    private fun levensteinDistance(x: String, y: String): Int {
```
Comment (ghemDD) [inline app/src/main/java/ch/epfl/sdp/blindwar/domain/game/GameHelper.kt:40]: Small typo *Levenshtein

**Review by ghemDD — APPROVED** [review]

File: app/src/main/java/ch/epfl/sdp/blindwar/domain/game/GameHelper.kt [hunk 5]
```diff
@@ -1,16 +1,12 @@
 package ch.epfl.sdp.blindwar.domain.game
 
-import kotlin.math.absoluteValue
-
 object GameHelper {
 
-    private const val keyboardRatio = 0.9
-    private const val vocalRatio = 0.75
-    private const val wordEpsilon = 50
+    private const val vocalCostMax = 5
```
Comment (ghemDD) [inline app/src/main/java/ch/epfl/sdp/blindwar/domain/game/GameHelper.kt:5]: Maybe use a ratio of the word length

**Pull request merged**